File: spacelox_vm/fixture/py_benchmark/list.py
import time

start = time.time()

for i in range(10000):
  collection = []
  for j in range(1000):
    collection.append(None)
    collection[j] = j

  sum = 0
  for j in range(1000):
    sum = sum + collection[1000 - j - 1]
    collection.pop()

print(time.time() - start)

File: laythe_vm/fixture/rb_benchmark/list.rb
start = Time.new

10000.times do |i|
  collection = []
  1000.times do |j|
    collection.push(nil)
    collection[j] = j
  end
    
  sum = 0
  1000.times do |j|
    sum = sum + collection[1000 - j - 1]
    collection.pop
  end
end

puts Time.new - start